File: make_table_breakpoints.m
function T=make_table_breakpoints(Break_result,param)
% table from break points
nb_param=length(param);
nb_col=8;
nb_row=height(Break_result);
Vartype={'string','datetime', 'double','double','double','double','double','double'};
VarNom={'param','Break_point','p_value','p_value_boot','level','minDiff','medDiff','maxDiff'};

T=table('Size',[nb_row nb_col],'VariableTypes',Vartype,'VariableNames',VarNom);
k=0;
for i=1:height(Break_result)
    if ~isempty(Break_result.station{i})
    for j=1:height(Break_result.results{i}.time)
        k=k+1;
    T.param(k)=Break_result.parameter{i};
     T.Break_point(k)=Break_result.results{i}.time(j);
      T.p_value(k)=Break_result.results{i}.pvalue(j);
      T.p_value_boot(k)=Break_result.results{i}.pvalue_boot(j);
      T.minDiff(k)=Break_result.results{i}.PrctDiff(j,1);
      T.medDiff(k)=Break_result.results{i}.PrctDiff(j,2);
      T.maxDiff(k)=Break_result.results{i}.PrctDiff(j,3);
       T.level(k)=Break_result.results{i}.level(j);

    end
    end
end

T=sortrows(T,'Break_point'); 

File: make_table_breakpoints.py
import pandas as pd
import numpy as np

def make_table_breakpoints(Break_result,model):
    rows = []

    for item in Break_result:
        station = item["station"]

        # Saltar si està buit
        if station is None or (isinstance(station, (list, str)) and len(station) == 0):
            continue

        results = item["results"]

        for j in range(len(results["time"])):
            row = {"param": item["parameter"],
                "Break_point": results["time"][j],
                "p_value": results["pvalue"][j],}

            if model == "Pettitt":
                row["p_value_boot"] = results["pvalue_boot"][j]

            row.update({"level": results["level"][j],
                "minDiff": results["PrctDiff"][j][0],
                "medDiff": results["PrctDiff"][j][1],
                "maxDiff": results["PrctDiff"][j][2]})

            rows.append(row)

    # Crear DataFrame
    T = pd.DataFrame(rows)

    # Ordenar per data
    T = T.sort_values(by="Break_point").reset_index(drop=True)

    return T